components/serializers.py

```diff
@@ -159,3 +195,70 @@ class Meta:
             "catalog",
             "controls_count",
         )
+
+
+class ComponentControlSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Component
+        fields = (
+            "pk",
+            "controls",
+            "description",
+        )
+
+    def update(self, instance, validated_data):
+        if controls := validated_data.get(
+            "controls"
+        ):  # and controls not in instance.controls:
+            existing_control = controls not in instance.controls
+            if not existing_control:
+                instance.controls = list(set(instance.controls).union(controls))
+
+            if description := validated_data.get("description"):
+                implemented_requirement = {
+                    "uuid": str(uuid.uuid4()),
+                    "props": [
+                        {
+                            "name": "security_control_type",
+                            "uuid": str(uuid.uuid4()),
+                            "value": "Hybrid",
+                        },
+                        {"name": "provider", "uuid": str(uuid.uuid4()), "value": "No"},
```

Comment:
Same question as above: is it intentional for each of these "uuid" fields to be different ids?

Reply:
Yes, these uuids are intentionally different. This is the reference material it is based on https://github.com/CMSgov/component-library/blob/main/django/oscal/django.json